include/wx/msw/ole/oleutils.h

```diff
@@ -66,18 +66,45 @@ inline void wxOleUninitialize()
 class WXDLLIMPEXP_CORE wxBasicString
 {
 public:
-    // ctors & dtor
-    wxBasicString(const wxString& str);
-    wxBasicString(const wxBasicString& bstr);
-    ~wxBasicString();
 
-    wxBasicString& operator=(const wxBasicString& bstr);
-
-    // accessors
-        // just get the string
+	// Takes over the ownership of bstr
+    wxBasicString(BSTR bstr = NULL);
+	
+	// Constructs the BSTR from wxString
+	wxBasicString(const wxString& str);
+    
+	// Creates a copy of the BSTR owned by bstr
+	wxBasicString(const wxBasicString& bstr);
+    
+	// Frees the owned BSTR
+	~wxBasicString();
+	
+
+	// Sets its BSTR to a copy of the BSTR owned by bstr
+	wxBasicString& operator=(const wxBasicString& bstr);
+
+	// Creates its BSTR from wxString
+	wxBasicString& operator=(const wxString& str);
+
+	// Takes over the ownership of bstr
+	wxBasicString& operator=(BSTR bstr);
+
+	// Returns the owned BSTR and gives up its ownership
+	BSTR Detach();
+
+    /// Returns the owned BSTR while keeping its ownership    
     operator BSTR() const { return m_bstrBuf; }
-        // retrieve a copy of our string - caller must SysFreeString() it later!
-    BSTR Get() const { return SysAllocString(m_bstrBuf); }
+     
+	// Returns the address of the owned BSTR
+	operator BSTR*() { return &m_bstrBuf; }
```

Comment:
`ByRef()` is supposed to imply that the parameter is passed by reference, i.e. it will be modified by the function. A more accurate name might be `AsWriteablePointer()`, but this is too long, I think.

The best, but also more complex, solution is IMHO to not have such public method at all, but rather have `wxBasicStringRef` helper class which would be friend of `wxBasicString`, and then use it like this:
```
wxBasicString bs;
CallSomeFunctionReturningBSTRAsOutParam(..., wxBasicStringRef(bs));
```
This is nice because it's safe: you can't just overwrite the internal `m_bstrBuf` accidentally. Of course, you can still do it, but you really have to go out of your way and, as a popular saying goes, the goal of a good library is to defend against Murphy, not Machiavelli.

And I don't like `GetAddress()` because this name doesn't imply at all that this method should be used only for calling functions returning `BSTR`, as I think ought to be the case.